File: launch_school/RB101-Foundations/small_problems/letter_counter.rb
=begin
-the problem requires that the string is broken into words array
-each word has to be counted for length
-each length is stored as key, and its count is stored as value
=end

def word_sizes(str)
  words_length_arr = str.split.map {|word| word.length}
  words_length_arr.each_with_object({}) {|num, hash| hash[num] = words_length_arr.count(num)}
end

puts word_sizes('Four score and seven.') == { 3 => 1, 4 => 1, 5 => 1, 6 => 1 }
puts word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 => 5, 6 => 1, 7 => 2 }
puts word_sizes("What's up doc?") == { 6 => 1, 2 => 1, 4 => 1 }
puts word_sizes('') == {}

File: launch_school/JS210-Fundamentals-of-Javascript/small_problems/easy_three/letter_counter.js
'use strict';

function wordSizes(string) {
  let words = string.split(' ');
  let letterCount = {};

  words.forEach(word => {
    let length = word.length;
    if (letterCount[length]) {
      letterCount[length] += 1;
    } else {
      letterCount[length] = 1;
    }
  });

  console.log(letterCount);
  return letterCount;
}

wordSizes('Four score and seven.');                       // { "3": 1, "4": 1, "5": 1, "6": 1 }
wordSizes('Hey diddle diddle, the cat and the fiddle!');  // { "3": 5, "6": 1, "7": 2 }
wordSizes("What's up doc?");                              // { "2": 1, "4": 1, "6": 1 }
wordSizes('');                                            // {}

function onlyWordSizes(string) {
  string = string.replace(/[^A-Za-z\s]/g, '');
  return wordSizes(string);
}

onlyWordSizes('Hey diddle diddle, the cat and the fiddle!');
